src/mongocrypt.c

```diff
@@ -606,27 +606,27 @@ static void _csfle_drop_global_ref(void) {
             old_state = g_csfle_state;
             dropped_last_ref = true;
         }
-    }
 
-    if (dropped_last_ref) {
```

Comment:
The reason for this boolean and copying the old state was to avoid doing this work within the protection of the mutex. If this is moved into that scope, then the code can be simplified to do the destruction directly without the state copying.